client/my-sites/plans-features-main/index.tsx

```diff
@@ -290,32 +291,44 @@ const PlansFeaturesMain = ( {
 	const intentFromSiteMeta = usePlanIntentFromSiteMeta();
 	const planFromUpsells = usePlanFromUpsells();
 	const [ forceDefaultPlans, setForceDefaultPlans ] = useState( false );
-
 	const [ intent, setIntent ] = useState< PlansIntent | undefined >( undefined );
+
+	const { segment: intentFromSegmentationSurvey, isLoadingSegment } = useSegmentedIntent(
```

Comment:
It looks like we are deriving the intent value according to the segmentation survey here. There are two main drawbacks here:

1. Once a site is created, the intent value will be stored in the stie meta. e.g. that's how a site created from `/setup/newsletter` will see the same plan mix at the logged-in `/plans`. Ideally all the survey data handling should only happen in the signup flow. After that, the site should just use the stored meta data.
2. An extra `GET` request to the survey endpoint will be fired everytime `<PlansFeaturesMain>` is rendered no matter it's needed or not, e.g. an existing account without ever seeing the survey visits `/plans`. 

I have a feeling that it might be something that can be addressed together with @chriskmnds 's [comment below](https://github.com/Automattic/wp-calypso/pull/90809/files#r1620213647) since it's likely also about pulling this to the consumer end. If not, it'd be wonderful if we can address as a follow-up :)

Reply:
Finally I see the light. Thank you so much @southp and @chriskmnds! Pushed.